doc/programming_guide/rendering.rst

```diff
@@ -162,21 +162,35 @@ are both contained in the ``WindowBlock`` uniform block. Which looks like this i
 
 You can view what uniform blocks exist in a :py:class:`~pyglet.graphics.shader.ShaderProgram` using the `uniform_blocks`
 property. This is a dictionary containing a Uniform Block name key to a :py:class:`~pyglet.graphics.shader.UniformBlock`
-object value. In the above example, the name would be ``WindowBlock`` while the ``window`` identifier is used in the GLSL
-shader itself.
+object value. In the above example, the name would be ``WindowBlock`` while the ``window`` instance identifier is used in the GLSL shader itself.
+
+Normally with OpenGL, you would have to manually assign a global binding point value to each Uniform Block for each Shader Program, as they are created. With Pyglet, the global binding values and assignments are all taken care of internally.
+
+Uniform Block's can be a convenient way to update uniforms of multiple Shader Programs at once, as their data is shared. This allows you to access the same information from multiple Shader Programs without having to bind every program using it, just to modify the uniform values. This can be achieved through a :py:class:`~pyglet.graphics.shader.UniformBufferObject`.
 
 To modify the uniforms in a :py:class:`~pyglet.graphics.shader.UniformBlock`, you must first create a
 :py:class:`~pyglet.graphics.shader.UniformBufferObject` using the
 :py:meth:`~pyglet.graphics.shader.UniformBlock.create_ubo` method.::
 
     ubo = program.uniform_blocks['WindowBlock'].create_ubo()
 
-The :py:class:`~pyglet.graphics.shader.UniformBufferObject` can then be used, and acts as a context manager for easy
-access to its uniforms::
+The :py:class:`~pyglet.graphics.shader.UniformBufferObject` can then be used as a context manager for easy
+access to update its uniforms::
 
         with ubo as window_block:
             window_block.projection[:] = new_matrix
 
+You can also create multiple :py:class:`~pyglet.graphics.shader.UniformBufferObject` instances if you need to swap between different sets of data. Calling :py:meth:`~pyglet.graphics.shader.UniformBufferObject.bind` will bind the associated buffer data to the Uniform Block.
```

Comment:
this sentence may read better as "will bind the buffer's data to the associated binding point". It should definitely talk about "binding point" instead of "Uniform Block".